src/hooks/useErc20.ts

```diff
@@ -144,17 +149,35 @@ export function useMultipleErc20(
 
     // Save to global state newly fetched tokens that are not null
     saveErc20s(fetched.filter(Boolean) as TokenErc20[])
+    if (fetched.includes(null)) {
+      setMaxAttemp((oldValue) => oldValue + 1)
+      setIndexProvider((oldValue) => (oldValue === XDAI_INFURA_ENDPOINTS.length ? 0 : oldValue + 1))
```

Comment:
As you mentioned in your proposal `2`, I agree that this should be at the provider level instead of here.
Besides, xdai endpoints don't make sense for rinkeby/mainnet, right?